src/filter/QueenSacrificeFilter.ts

```diff
@@ -0,0 +1,106 @@
+import { VideoFilter } from "@/model/VideoFilter";
+import { Video } from "@/model/Video";
+import { Pgn } from "@/model/Pgn";
+import { usePgnsStore } from "@/stores/pgnsStore";
+
+/* 
+  There are some cases where a queen sacrifice may be wrongly detected:
+  - A blunder where the queen is captured without any compensation. However, it is unlikely that such games would be analyzed on the channel.
+  - A queen exchange combination, such as a check in-between move before recapturing the other queen.
+
+  There are also some cases where the queen sacrifice may not be detected:
+  - This function does not currently handle the scenario in which a queen exchange takes place and, later, a pawn is promoted to a queen, and then sacrificed.
+*/
+function isQueenSacrifice(pgn: string): boolean {
+  const moves = pgn
+    .replace(/\d+\.\s+/g, "") // Remove move numbers
+    .trim()
+    .split(/\s+/); // Split by spaces
+
+  const captureRegex = /([QRNB])?([a-h])?x([a-h][1-8])?/;
+
+  let whiteQueenPosition = "d1"; // White Queen starts at d1
+  let blackQueenPosition = "d8"; // Black Queen starts at d8
+
+  const lastMoveIndex = moves.length - 1;
+
+  for (let i = 0; i < moves.length; i++) {
+    const move = moves[i];
+
+    const captureMatch = move.match(captureRegex);
+
+    // If it's a capture move
+    if (captureMatch) {
+      const targetSquare = captureMatch[3]; // The square where the capture happens
+
+      const isWhiteQueenCapture = targetSquare === whiteQueenPosition;
+      const isBlackQueenCapture = targetSquare === blackQueenPosition;
+
+      if (isWhiteQueenCapture) {
+        if (captureMatch[1] === "Q") {
+          // If it's a queen capturing a queen, it's an exchange, not a sacrifice
```

Comment:
Actually it is not. It is only exchange if there will be immediate exchange on same field.

Consider:
`20. Qxd5 /Black queen capture/ h6 /random pawn move by black/ 21. Qd5 /White queen is stil in game/`

To handle this situation you would have to check if next move is capture on same field.

Reply:
True, and re-capture of queen on next move check is part of original logic.